### ISSUE TITLE
Is archzfs done?

### ISSUE THREAD

**Issue by johnramsden** [archzfs/archzfs#545]
There have been no new packages in months, even for the lts. Should people look for alternative solutions, or is something holding back a release?

**Comment by fryfrog** [comment]
I was just coming here to ask if @minextu was on a long vacation. Might be time to try the [aur/zfs-dkms](https://aur.archlinux.org/packages/zfs-dkms) package again. :|

**Comment by jrudess** [comment]
Yes, at this point, unless someone has the bandwidth to fork this repo and take over ownership, I would consider it done.

**Comment by n0xena** [comment]
well - building the packages works smoothly
currently there seems an issue with fakeroot or something like this as the building for 6.10.5 works but the final packaging fails with some weird ssl related issue - have a look at https://github.com/archzfs/archzfs/pull/542#issuecomment-2293366571
so it's not so much a problem of using this repo to build the packages your own - currently just the automated buildbot doesn't work - but that was already the case several times in the past and at least until now the maintainer usually fixed it rather fast
I wouldn't call this repo dead - as pretty much all arch source refer to it - even the wiki just links here with no information how to build a pkg from source your own - which is all this repo here does: a few shell scripts to build complete pkgs so one can use pacman to keep them consistent
you also could go yolo and just run the usual
./configure
make
make install
but this will likely fail according to what I understand from the scripts as they also use autogen and quite a lot of configure options
so - if at all - it's not this repo which may require a takeover or for - but rather the archwiki to get extended with a section on how to build the packages "by hand" and maybe even without clean-chroot or such

**Comment by jbrodriguez** [comment]
i've also gone the linux-lts route, which openzfs will very likely support without major issues

as for bulding zfs-linux-lts, in addition to using this repo, i've also found some other ways, including the one i settled for: a docker to build them (https://jbrio.net/posts/5-ways-archlinux-zfs/)

but, hopefully minextu can get back in the game, maybe he's no longer into archlinux or openzfs, maybe he has no bandwidth for this or has other life priorities 🤷‍♂️

**Comment by soi1uhtgnei2urtonwe4u** [comment]
>There have been no new packages in months, even for the lts. Should people look for alternative solutions, or is something holding back a release?

>but, hopefully minextu can get back in the game, maybe he's no longer into archlinux or openzfs, maybe he has no bandwidth for this or has other life priorities

v2.2.5 came out on the 6th, that's more than 2 weeks now. AFAIK nothing is holding the release back (the recent kmod issue looks [fixed](https://gitlab.archlinux.org/archlinux/packaging/packages/kmod/-/commit/5c47a72852d7dc90d0f378e2b174be7e33a54abe) now too (commit reverted)). No v2.2.5 release for lts could mean the maintainer was afk for the usual holiday length of 10-14 days (no commits indeed in that time). The alternative solution would be to use the AUR.

@minextu Did no commits for 10 days, but then has been doing every day commits from 19.08 to 22.08 into a private repo again. If he no longer maintains then he should mention it.

I think he was afk a bit and I expect the 2.2.5 release within the next days/week. 
PS: Yesterday Linux 6.10 compat was [pushed](https://github.com/openzfs/zfs/commit/9e15877dfb3e80021551301aac71976216b3fe1b), but v2.2.5 officially supports only Linux 6.9.

**Comment by rearden888** [comment]
I hope not.  I think the framework for this works well, it just needs someone to continue shepherding it as versions update.  Not sure if @minextu had other requirements going on, but maybe they'd be willing to add other committers to the repository to help out if they aren't available?

**Comment by ArnCo** [comment]
I have proposed to help multiple times here... I will probably switch back to the AUR if this repo dies.

**Comment by rearden888** [comment]
I'm hoping to move the conversation on the state of this project and possible avenues forward to this topic, vice in the current PR for OepnZFS 2.2.6.  If we were to want to actually fork this, it's probably important to note that the archzfs.com domain would definitely *not* come along.  We'd definitely need to ask to mark the project orphaned per [here](https://wiki.archlinux.org/title/Arch_User_Repository#Flagging_packages_out-of-date) from the Arch Wiki.  Once it's marked Orphan, another AUR user can ask to adopt it and thereby upload a new PKGBUILD.  The Arch Wiki Page [zfs](https://wiki.archlinux.org/title/ZFS) will need to be updated, along with the definition for the archzfs user repository entry [here](https://wiki.archlinux.org/title/Unofficial_user_repositories#archzfs)  .

Finally, the infrastructure would need to be updated.  Now in this case, we're talking about forking on github, updating the package signing key, ensuring it's distributed and linked, starting an alternative CI server, as well as seed mirror for the repository.  

Anyway, I think it's worth seeing if @minextu is still wiling to potentially bring on a subset of active users as maintainers to help speed up updates.  I think it's worth waiting a few more days to get some response.

**Comment by fryfrog** [comment]
I actually emailed @minextu at the address I found via (I think?) the AUR, but haven't gotten a reply. :(

This isn't an AUR package, so really "forking" it would just be cloning the repo and getting it all setup and working again and then updating all the wiki details. Maybe waiting for archzfs.com domain to expire and not renew?

**Comment by rearden888** [comment]
> I actually emailed @minextu at the address I found via (I think?) the AUR, but haven't gotten a reply. :(
So have I, haven't heard anything, either.

> 
> This isn't an AUR package, so really "forking" it would just be cloning the repo and getting it all setup and working again and then updating all the wiki details. Maybe waiting for archzfs.com domain to expire and not renew?

I certainly agree that I think most people access this through the unofficial repo.  There is a AUR PKGBUILD though, so it's probably worth keeping it updated.  It appears to mostly be a matter of changing version numbers.  

We could certainly also wait for the archzfs.com domain to expire, but who knows if it's on auto renewals or something similar.

**Comment by fryfrog** [comment]
Are you talking about [aur/zfs-dkms](https://aur.archlinux.org/packages/zfs-dkms) for example? That is a different package from archzfs's package. If you're not, what are you talking about?

And to be clear, I mean roughly fork it *now* and then if the domain expires some time in the future, take it over to help everyone who doesn't know it changes. I wouldn't hold a fork off for the domain. Could maybe even get archzfs.org or net or something if available. :)

**Comment by endreszabo** [comment]
Fortunately, I asked for delegation of kernels.archzfs.com so I can change the records for that anytime. Sadly, it is not the case for the apex archzfs.com itself of course.

I can still "publish" the dependant kernel packages under kernels.archzfs.com. Or any other domains we may come up with later on.

**Comment by rearden888** [comment]
> Are you talking about [aur/zfs-dkms](https://aur.archlinux.org/packages/zfs-dkms) for example? That is a different package from archzfs's package. If you're not, what are you talking about?
> 
> And to be clear, I mean roughly fork it _now_ and then if the domain expires some time in the future, take it over to help everyone who doesn't know it changes. I wouldn't hold a fork off for the domain. Could maybe even get archzfs.org or net or something if available. :)

While I don't think it's used much, there *is* a PKGBUILD listed and served under the zfs-linux package in the AUR [here](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zfs-linux)  Probably not the biggest deal to change, but it is listed there.

Also, yes, I understand this project is intended to be used as a user Pacman repository, which is what archzfs-ci builds and deploys.

**Comment by rearden888** [comment]
> Fortunately, I asked for delegation of kernels.archzfs.com so I can change the records for that anytime. Sadly, it is not the case for the apex archzfs.com itself of course. I can still "publish" the dependant kernel packages under kernels.archzfs.com. Or any other domains we may come up with later on.
> […](#)
> On Mon, Sep 09, 2024 at 21:00:46-0700, Donald Webster wrote: Are you talking about [aur/zfs-dkms](https://aur.archlinux.org/packages/zfs-dkms) for example? That is a different package from archzfs's package. If you're not, what are you talking about? And to be clear, I mean roughly fork it *now* and then if the domain expires some time in the future, take it over to help everyone who doesn't know it changes. I wouldn't hold a fork off for the domain. Could maybe even get archzfs.org or net or something if available. :) -- Reply to this email directly or view it on GitHub: [#545 (comment)](https://github.com/archzfs/archzfs/issues/545#issuecomment-2339560416) You are receiving this because you are subscribed to this thread. Message ID: ***@***.***>

I personally feel bad forking this without at least some answer from @minextu  If you have already discussed and achieved delegation on  subdomain from him, do you have any better way to get ahold of him?  I've only rarely received responses.

**Comment by aereaux** [comment]
> > Are you talking about [aur/zfs-dkms](https://aur.archlinux.org/packages/zfs-dkms) for example? That is a different package from archzfs's package. If you're not, what are you talking about?
> > And to be clear, I mean roughly fork it _now_ and then if the domain expires some time in the future, take it over to help everyone who doesn't know it changes. I wouldn't hold a fork off for the domain. Could maybe even get archzfs.org or net or something if available. :)
> 
> While I don't think it's used much, there _is_ a PKGBUILD listed and served under the zfs-linux package in the AUR [here](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zfs-linux) Probably not the biggest deal to change, but it is listed there.
> 
> Also, yes, I understand this project is intended to be used as a user Pacman repository, which is what archzfs-ci builds and deploys.

Just chiming in to say that I do use the PKGBUILD for zfs-linux-lts.  I have been manually updating the versions when linux-lts gets updated, and this project provides value for me for the tests that are run to validate each version.

**Comment by stevleibelt** [comment]
@rearden888 and just to all others,

any Idea how to proceed? On the one hand, I would like to give @minextu enough time to response, on the other hand, I would also like to get things rolling.

Best approach, from my point of view, would be that @minextu would be able to add more trusted maintainers to keep things simple.
But back to the timedelta, what about keeping a month to response for @minextu open and than do steps that are communicated in this issue?

Don't get me wrong, I am happy about any maintainer here but I also try to not be so rude to anybody who had served the community so well for so long.

**Comment by Krakonos** [comment]
Talking as a maintainer that doesn't always respond promptly, the following is (in my opinion) very reasonable course of action:

I'm not sure how much work it actually is, but i would suggest to create a (possibly temporary) fork to bridge the current situation and make extra clear this is not a "hostile takeover" and a better solution, in form of adding more maintainers to the project (and possibly domain delegation etc.) is strongly preferred. The fork could possibly be a new repository for binaries (TBH, just switching out the repo under his hands seems a bit sketchy - and people not reading this thread might be surprised, possibly even suspect an active attack on the repository. In this regard, I think it should be active choice to switch to the new repo).

For any decent project, having some redundancy is beneficial and can actually relieve the stress on the main maintainer as well. Life can happen. Family issues, work might require 25 hours a day for some time, stuff happens.

@minextu Whatever the reasons you might have, I wish you the best and sincerely hope you're not facing serious issues. In case there is something (even remotely possible) I or the community could do to help out, please reach out.

**Comment by endreszabo** [comment]
> If you have already discussed and achieved delegation on subdomain from him, do you have any better way to get ahold of him? I've only rarely received responses.

I wish I had any better contact options. We discussed about the delegation right here in [Issue #368](https://github.com/archzfs/archzfs/issues/368#issuecomment-720074834).

**Comment by rearden888** [comment]
Yeah, to be clear, the last thing I want to do is fork this.  That's literally the hardest solution to this problem, and requires the most amount of upheaval, and has the most downside should @minextu come back and want to be involved again.  The best option is getting a few interested and capable individuals access to maintain the repos and servers.

**Comment by endreszabo** [comment]
Someone living near Dortmund should visit and ring all 8 doorbells of his [Stiegenhaus](https://maps.app.goo.gl/gLhHvvgk9xcPxHoS7).

**Comment by endreszabo** [comment]
> Someone living near Dortmund should visit and ring all 8 doorbells of his [Stiegenhaus](https://maps.app.goo.gl/gLhHvvgk9xcPxHoS7).

Of course, this might be incorrect, I found this address on the [impressum page of one of his side projects](https://nexttrex.de/Impressum). There's also a different e-mail address.

**Comment by johnramsden** [comment]
Maybe we shouldn't doxx the guy

**Comment by rearden888** [comment]
Yeah, I agree.   Can you take down the link to the address and links to it down @endreszabo? I realize its potentially publicly available, but probably not a great idea to have it up in such a conspicuous location.

**Comment by n0xena** [comment]
As my fork was mentioned: Please note, that all I did was to use Github's function to get a "copy of my own" to apply the required updates, namely the update to zfs-2.2.5/.6 and the fix for the break of kmod-33, and just upload the resulting packages back to github.
I by no means am a dev or maintainer nor do I really understand what I'm doing - I'm just a stranger using others' work and provide back the result. I'm aware of the current situation and read several issues upstream that there are several problems even on lts-6.6.x and that ZFS itself currently is rather unstable.
I don't take any responsibility for any issues resulting in using my builds - everyone can do the same. I just try to give something back to the community I profit from.

As already mentioned: minextu did a great job of setting up this repo. I still criticize that there're not much more information how to build a proper arch package and all sources like the wiki just link back here. The upstream makefile only provide RPM and DEB packages - with an issue going on that DEB packages seem to be faulty since ZFS-2.x. So maybe the ZFS maintainers could be requested to provide something for arch upstream - like either how to build it proper your own or to integrate a pkgbuild into the main makefile. At least it would be nice to find anything in the arch wiki. Arch is one of the major base distributions when sorted by package type / manager - it's not just a meme distribution - hence upstream may consider adding official support and builds for it.

btw: currently I upload zen and lts on request from users over at the arch bbs - I only use the std package myself - if someone wants additional builds like hardened, vfio or archiso please just request them and I will provide them

**Comment by jbrodriguez** [comment]
> several problems even on lts-6.6.x

where have you seen those issues for lts ? i checked open issues on the openzfs (https://github.com/openzfs/zfs/issues) and don't seen mentions of linux 6.6.x

**Comment by fryfrog** [comment]
The AUR version is also missing the hooks to allow unlocking an encrypted root pool via ssh.

**Comment by lnxsrt** [comment]
So what's the gameplan here? How much longer are we waiting? A fork really isn't a huge deal and could easily be merged back into this project when/if minextu returns.

**Comment by n0xena** [comment]
My thoughts are on ZFS upstream: They offer building DEB and RPM but not PKG.
Arch is one of the main base distributions and has a strong user base. It should be covered by ZFS itself.
Otherwise someone should analyze the configure options chosen by minextu and should extend the archwiki so it includes specific step-by-step workflow how to create arch packages from what ZFS provides in thier repo. Maybe even setup some new automation.
To put it this way: We as the community should work towards spreading the knowledge how to build a package rather than burden a few individuals with keeping it up. This way anyone want to use ZFS on Arch can look it up at wiki with repos like this provide additional AUR or pre-built.

I don't mind to provide pre-built packages to help the community - but neither can I tell for how long I will keep this up nor if I'm able to maintain any patches if the current workflow somehow breaks. Hence I use Github instead of like GoogleDrive or similar: So anyone can trace what I modified (or rather: that I not modified anything but the updates and kmod fix amd the usernsme) to provide some level of trust that the provided packages result from the provided codebase. And so that anyone who does not trust can just clone it and build the packages on thier own.

**Comment by neworld** [comment]
The step-by-step guide is encoded well in the AUR package. This repo does more than build packages. It ensures, at some level, compatibility with the newest kernel version. I always use DKMS and spend countless hours restoring the system to an encrypted root because the kernel version was unsupported.

**Comment by n0xena** [comment]
> The step-by-step guide is encoded well in the AUR package. This repo does more than build packages. It ensures, at some level, compatibility with the newest kernel version. I always use DKMS and spend countless hours restoring the system to an encrypted root because the kernel version was unsupported.

I'm not quite sure we're on the same page about the step-by-step stuff. The AUR .git files all only link back here to this repo. It than regard it's not different from as cloning this repo in its entirety and execute ./build.sh.
My point about "step-by-step" is more like this:

zfs-utils
```
prepare() {
    cd "${srcdir}"/zfs-${pkgver}

    # pyzfs is not built, but build system tries to check for python anyway
    ln -sf /bin/true python3-fake

    autoreconf -fi
}

build() {
    # Disable tree vectorization. Related issues:
    # https://github.com/openzfs/zfs/issues/13605
    # https://github.com/openzfs/zfs/issues/13620
    export CFLAGS="$CFLAGS -fno-tree-vectorize"
    export CXXFLAGS="$CXXFLAGS -fno-tree-vectorize"

    cd "${srcdir}"/zfs-${pkgver}

    ./configure --prefix=/usr \
                --sysconfdir=/etc \
                --sbindir=/usr/bin \
                --with-mounthelperdir=/usr/bin \
                --with-udevdir=/usr/lib/udev \
                --libexecdir=/usr/lib \
                --localstatedir=/var \
                --with-python="$PWD/python3-fake" \
                --enable-pyzfs=no \
                --enable-systemd \
                --with-config=user
    make
}
```
zfs-linux
```
build() {
    cd "${srcdir}/zfs-${_zfsver}"
    ./autogen.sh
    ./configure --prefix=/usr --sysconfdir=/etc --sbindir=/usr/bin --libdir=/usr/lib \
                --datadir=/usr/share --includedir=/usr/include --with-udevdir=/usr/lib/udev \
                --libexecdir=/usr/lib --with-config=kernel \
                --with-linux=/usr/lib/modules/${_extramodules}/build \
                --with-linux-obj=/usr/lib/modules/${_extramodules}/build
    make
}
```
why minextu specifically chose those ./configure options? why did they chose to use autoreconf for the utls but ./autogen.sh for the kernel module?
I just did a build of Linux From Scratch - and by this only understand a few of the options. But my point is: Everyone relies on the fact that these are the best ones - based on WHAT? I couldn't find any explanation here in this repo nor on the archwiki nor over at zfs upstream - but I trust minextu that they did a good job - as it works.
What I like to see, best upstream, is a guide which explains these options and the workflow and why do I have to use these and not others - along with: How do I all pack this into an arch PKG?
Only the ZFS maintainers know best how thier autoconf and configure really work and what options to set to build the utils packages or one of the kernel modules - and as mentioned they do provide it for RPM and DEB (aside from DEB broken since 2.x). So I see it fit to request them either add arch PKG upstream or provide info how someone could end up with the result minextu provided us with here - because THIS information I wasn't able to find.
If anyone is smarter than me and can provide such explanations what the options do and why they're chose - please do a fork and add the documentation. I'm happy to merge it into my fork.

**Comment by sudoBash418** [comment]
> why minextu specifically chose those ./configure options?

Most of these options (`--prefix`, `--libexecdir`, etc.) are given to ensure the resulting package follows the [Arch package guidelines](https://wiki.archlinux.org/title/Arch_package_guidelines), particularly the [directories section](https://wiki.archlinux.org/title/Arch_package_guidelines#Directories). Other options (such as `--with-linux`) may not be explicitly mentioned on the Arch Wiki, but are based on the way that Arch handles kernel packages.

> What I like to see, best upstream, is a guide which explains these options and the workflow and why do I have to use these and not others - along with: How do I all pack this into an arch PKG?
Only the ZFS maintainers know best how thier autoconf and configure really work and what options to set to build the utils packages or one of the kernel modules - and as mentioned they do provide it for RPM and DEB (aside from DEB broken since 2.x). So I see it fit to request them either add arch PKG upstream or provide info how someone could end up with the result minextu provided us with here - because THIS information I wasn't able to find.

In other words, these are mostly Arch-specific aspects of the package that would explicitly *not* be the responsibility of the upstream ZFS developers to maintain.

> why did they chose to use autoreconf for the utls but ./autogen.sh for the kernel module?

I'm not 100% sure myself, but keep in mind that these two packages were originally submitted and are currently maintained by separate sets of people.

For `zfs-utils` specifically, it looks like it was [added when](https://aur.archlinux.org/cgit/aur.git/commit/?h=zfs-utils&id=c8148ca3887ff74614d8c35337a126b559926feb) the package temporarily switched to tracking a git branch, and was never removed when they switched back to a tarball release (btw, `git blame` is very useful for answering these sorts of questions).

**Comment by stevleibelt** [comment]
> So what's the gameplan here? How much longer are we waiting? A fork really isn't a huge deal and could easily be merged back into this project when/if minextu returns.

Just to give some feedback and bring up a suggestion. What about waiting for @minextu reply until first of October?
But only **if** someone is capable of providing a working and stable alternative.

**Comment by neworld** [comment]
> I'm not sure we're on the same page about the step-by-step stuff.

[zfs-dkms AUR package](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zfs-dkms) is building from official repository. 

My two cents. Even if I have some skills to understand and build ZFS by myself, I am still keen to use something maintained and tested by the wider community. @minextu did an incredible job; no instructions could replace his effort. I want to utilize ZFS in my daily life as a user.

> Everyone relies on the fact that these are the best ones - based on WHAT?

Not me. I am relaying the tested solution on many machines by many professionals, but not on having the best possible configuration. Also, upstream is constantly changing, and instructions usually become outdated unless someone disciplined maintains them.

**Comment by nerzhul** [comment]
so what's the plan, tbh i prefer this packaging over zfs dkms. Cannot some people ask github for maintainer on this repo, i know they can do this if needed and the maintainer disappeared and there is trust in someone new

**Comment by n0xena** [comment]
well - there's

> Maintainer: Kevin Stolp <kevinstolp@gmail.com>

as aur-maintainer of zfs-dkms and zfs-utils - I'm not sure if he has an eye on this repo here - but it would be easier to ask the aur staff to give him maintainer rights over at the aur than try to request Microsoft to add a random maintainer to someone elses github repo

**Comment by rearden888** [comment]
> so what's the plan, tbh i prefer this packaging over zfs dkms. Cannot some people ask github for maintainer on this repo, i know they can do this if needed and the maintainer disappeared and there is trust in someone new

Many of us have emailed him multiple times, haven't gotten a response.  I think I'm going to email one more time.  The process of forking/re-basing is not trivial, plus we will not have access to the archzfs.com domain, since that is his personal property.

**Comment by n0xena** [comment]
just as an update: 6.11.1 build is available: https://github.com/n0xena/archzfs/releases/tag/2.2.6-6.11.1
works for me so far - although please note the linked question hence I keep up the 6.10.10 for a while

**Comment by stevleibelt** [comment]
@rearden888 and for all others, if there will be a fork, email me so we can figure out if and how I can support.

@n0xena thanks!

**Comment by s-i-e-v-e** [comment]
> I'm still for some collaborative work to update the archwiki page with enough information everyone can just grep the upstream openzfs repo and build the utils and kernel package for themselfs but in a consistent manner

This is the best option, IMO. Getting upstream to add a new artifact, or getting people to agree to switch to a new provider  (even if it is in their best interest) is an uphill talk. See the "waiting for minextu" game being played in this very thread, for instance.

My heartfelt thanks to minextu for all his years of service to the community, BTW.

> ZFS
I am on year 12 of ZFS on Linux and have never lost data. But that is a sample size of one. Getting mirroring right is important.

>  I'll try to build a package for it
Last month, I tried doing it in a docker/podman container using parts from the AUR scripts. While the artifacts were built properly, two sets of warnings were emitted in the process. While I don' recollect the first set, the second was related to padding issues in the encryption code, and I did read a ZFS maintainer's comment that they can be ignored.

Can I ask what your build experience has been? Do the artifacts get built without emitting ANY warnings?

**Comment by s-i-e-v-e** [comment]
@n0xena

> I'm still for some collaborative work to update the archwiki page with enough information everyone can just grep the upstream openzfs repo and build the utils and kernel package for themselfs but in a consistent manner

This is the best option, IMO. Getting upstream to add a new artifact, or getting people to agree to switch to a new provider  (even if it is in their best interest) is an uphill talk. See the "waiting for minextu" game being played in this very thread, for instance.

My heartfelt thanks to minextu for all his years of service to the community, BTW.

> ZFS

I am on year 12 of ZFS on Linux and have never lost data. But that is a sample size of one. Getting mirroring right is important.

>  I'll try to build a package for it

Last month, I tried doing it in a docker/podman container using parts from the AUR scripts. While the artifacts were built properly, two sets of warnings were emitted in the process. While I don' recollect the first set, the second was related to padding issues in the encryption code, and I did read a ZFS maintainer's comment that they can be ignored.

Can I ask what your build experience has been? Do the artifacts get built without emitting ANY warnings?

**Comment by n0xena** [comment]
> Can I ask what your build experience has been? Do the artifacts get built without emitting ANY warnings?

Actual there're only two things one has to look out for:

1) Do follow the build instructions! Step 1 is very important: open conf.sh and change the makepkg_nonpriv_user to match your actual username. As minextu's first name seems to be Jan he named his user "jan". In my fork I just changed that to match my local user just for the sake of convenience.
2) For some reason the initial setup of the clean-chroot-manager directory /scratch has to be done several times with the initial `./build.sh utils update make` so the utils package gets built and installed correctly so the kernel modules can be build.

Other than this it's as simple as `time MAKEFLAGS="-j$(nproc)" sudo ./build.sh all update make` and everything gets done in about 10 minutes.

The terminal spits out a few warnings here and there I don't understand - overall I only have expeirence in Java and started learning C and C++ last summer - but not everything gets build: As you can see from my release I currently get done the utils package and the modules for standard kernel, hardnened, lts, zen and dkms. The iso and vfio packages gets skipped with some older versions no longer found - but I guess that could be fixed easy by updating the versions the log complains about.

**Comment by s-i-e-v-e** [comment]
Okay, so you are using the build scripts from archzfs as is.

I only wanted zfs-linux and zfs-utils, so I went about working off of the AUR scripts. Putting it in a podman/docker container lets me generate the artifacts without polluting my system with random dependencies.

But I didn't package the output even after a successful build _because_ of the warnings.

Will try it again and post a link to a repository with the complete build script that others can try with podman/docker. Ideally anyone should be able to do it on their system in a few minutes without fiddling with usernames etc.

**Comment by jbrodriguez** [comment]
@s-i-e-v-e there are already docker based build systems, so you don't need to reinvent the wheel

- https://github.com/jbrodriguez/arch-zfs-docker (by me)
- https://github.com/markusressel/arch-zfs-docker (a fork of my repo that is more feature complete)

**Comment by Doridian** [comment]
Thought I'd chime in here after I got confused why my servers can't take any more kernel updates with archzfs.

I do not have much I can contribute to this discussion specialized on archzfs, but I can share my thoughts on the more general infrastructure.

Mainly toward the point of:

> I've been thinking if it wouldn't actually be best to have everything set up here, on GitHub, like recently [LizardByte did with Sunshine](https://github.com/LizardByte/pacman-repo).

I think setting everything up on GitHub might be a great option, for the reasons you mentioned. Unless archzfs suddenly sees a huge rise in popularity (and even then), I doubt GitHub will have any issues distributing those packages.

The only possible downside I can see would be if GitHub decides to cease existing, but so could any other hosting location, and if everything (all the code + config) is in Git, at least it should be somewhat easy to port to another platform, as opposed to bespoke infrastructure.

I would expect GitHub's CI runners to be good enough to build archzfs (which would make the build process completely transparent, if that is a goal). Depends how fast the build needs to be, given full kernel recompilation and all that, but...
Worst case that is also easy to fix by hosting your own GitHub runner, which would also make that component easy to swap out if the "current host" at the time decides to no longer provide those services.

**Comment by kerberizer** [comment]
GitHub Actions are free for the standard GitHub-hosted runners in public repositories, so I wouldn't expect problems here. Build times might not be the best, indeed, but they should still be within reasonable expectations. And if the need arises, we could really use self-hosted runners as well. As for the repo, I found it curious how @LizardByte had implemented it via the [GitHub releases](https://github.com/LizardByte/pacman-repo/releases). But it works and also makes sense, since there are [no quotas on releases](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases#storage-and-bandwidth-quotas). And we could still provide mirrors, like there are for the current repo, if we decide to go down the all-GitHub route.

**Comment by Doridian** [comment]
> As for the repo, I found it curious how https://github.com/LizardByte had implemented it via the [GitHub releases](https://github.com/LizardByte/pacman-repo/releases).

Yeah, using a "static" release that is overwritten is a practice I have seen (and used) in the past. It works quite well.
Especially since the Arch file naming pattern seems to work out well here as well.

If mirrors are needed, I can gladly offer some space on my server which already runs a full Arch mirror anyway, so that would add practically nothing to that server's load.

Only difference I'd make from the Sunshine repo is to sign the packages via the GitHub Actions runner (put the signing key in a "Secret" only accessible from the "production" branch, aka `master` probably)

**Comment by n0xena** [comment]
I'm not sure about ZFS upstream but I'd like to see some backup repo on another platform just for redundancy. I doubt that Microsoft would kill ZFS or archzfs - but even GitHub can have issues.

**Comment by Doridian** [comment]
Out of boredom I have started to poke at a possible GitHub Actions implementation: https://github.com/archzfs/archzfs/pull/551

Missing a lot of features for now, but I'll just work on those time goes (also happy for feedback of course)

**Comment by stevleibelt** [comment]
> I'm not sure about ZFS upstream but I'd like to see some backup repo on another platform just for redundancy. I doubt that Microsoft would kill ZFS or archzfs - but even GitHub can have issues.

Depending on the size, I could host something on my private server.
Alternative there is [codeberg.org](https://join.codeberg.org/), [gitlab.com](https://about.gitlab.com/) or maybe - but this would take time - I could ask my local university if they could offer some space. They already sponsor a [Tier 2 mirror](https://archlinux.org/mirrors/informatik.tu-freiberg.de/).

**Comment by n0xena** [comment]
Don't get me wrong - it's nothing against Microsoft - but remember when Facebook disconnected itself from the internet when they misconfigured their routers? And wasn't there some similar issue with Cloudflare? Point is: These big tech giants with thier own infrastructure fuck up from time to time - and although often it gets resolved rather quick it shows that rely on just one service can and often will bite you. It's just meant as a 2nd source when the primary becomes unavailable.

**Comment by kerberizer** [comment]
> It's just meant as a 2nd source when the primary becomes unavailable.

I understand these concerns. Considering that OpenZFS itself is hosted here, if GH goes down, not having ArchZFS available would probably be the least of our troubles, but I agree that having at least one more "authoritative" source of our work would be a good idea.

I've started setting up https://gitlab.com/archzfs to mirror the repos. With time, we might also replicate the CI system there as well. If used merely as a backup, the free-tier limits should be fine (and we can also use self-hosted runners if needed). GitLab also offers the advantage that if one day our project becomes important or popular enough, who knows, it might even be adopted on the official Arch's GitLab.

As for having a backup for the already built packages, continuing the practice of having mirrors of the main repo should likely do the job.

As always, I really appreciate all suggestions and concerns!

**Comment by rearden888** [comment]
Well, given we're now nearly a week past the 1 Oct date and we've not heard anything from Jan, I think it's time to start the process of forking and continuing.  I have the means to run and maintain a VPS build server that could run the docker containers and at least get the project building again.  I already host a US Mirror of the archzfs packages (zxcvfdsa.com).  

I agree a GH derived build solution will likely be preferable for the long run, but I think in the short term, simply running the existing, working infrastructure makes sense for a more rapid RTS.

In service of this, I propose we start a new project on GH and fork all the current ArchZFS repos there.  I've termed this project (at least in my head)  "Archzfs-ng.".  I'm traveling right now, but when I reach my destination, I plan to see what's involved in that and potentially start setting it up.  

I will also go to the AUR and submit the request to ask to mark the project abandoned and see about gaining control.

Also, I think we should consider meeting in some sort of real time/semi-real time venue, ie Discord of Matrix.  I run a small Matrix server for myself (matrix.zxcc.gg) and I'm happy to start a room there and post it for people to join and discuss.  If Discord is preferred, that's fine too, I think it's pretty easy to set up a "server" there and start discussions.  I'm also certainly open to adding collaborators from the start so as to avoid this sort of single point of failure situation in the future.

**Comment by kerberizer** [comment]
It's perfectly fine if you prefer forking the repos—this is the nature of open-source projects, after all.

That said, I'll continue focusing on setting up the GitHub-based (and possibly GitLab-based) build system and repo. @Doridian has already made good progress in #551, so that’s the direction I’d like to pursue.

My primary goal remains to develop a resilient, community-driven solution that doesn't rely on any single person or infrastructure. This ensures the long-term stability of the project and avoids any single points of failure.

Having a place to chat is certainly a good idea. Matrix is fine, though I’m not sure how widely adopted it is. To cover more ground, I’ve created a Discord server in the meantime:

Permanent invite: https://discord.gg/TZsb3cwdUM

**Comment by Doridian** [comment]
> @Doridian has already made good progress in #551, so that’s the direction I’d like to pursue.

Yeah, it seems to be working. I tested installing from the "release repo" and it seemed to install fine in a Docker container.
Feel free to change the naming, I've thrown around `master` and `mainline` and `Release` and `latest` a fair bit in those YML files, because I'm not sure which term is best used and where, but yeah.

If you need any help or got questions or requests, feel free to let me know (either here, PR or Discord). This goes for anyone reading this also, I'm glad to answer questions, time permitting

**Comment by stevleibelt** [comment]
@rearden888 

what about `archzfs-ce` to show that it is not a one man show anymore but a community edition?
But I am also fine with `archzfs-ng` :-). I've just asked myself what I would expect from a `next generation` archzfs :smile: 

Anyways, to give you some feedback, I vote for matrix and again, if needed and possible, I would like to be a part of this journey as I am as end user working with zfs on arch linux since it was a pure fuse project.
I can support even the build phase later on, but can not support this because of missing knowledge right now.

**Comment by Doridian** [comment]
I have managed to work out a fairly simple mirroring shell script: https://github.com/archzfs/archzfs/blob/8baf71c3495ce6754efc03aa1abe8c8bca071404/mirror.sh (also part of the PR now).

This minimizes downloads by using the GitHub API's file modification timestamps.
It uses a temp folder mainly so it is able to delete files no longer present in the release easier.

**Comment by kerberizer** [comment]
Outstanding work, @Doridian, thank you so much indeed! It's already late night here, but I'll try reviewing and merging #551 as soon as possible.

In the meantime, could I please ask everyone to test Doridian's work (obviously, please don't do it on production or important systems): https://github.com/Doridian/archzfs/releases/tag/latest

**Comment by demizer** [comment]
Howdy folks. I started this repo many years ago. I still have ownership and can transfer that to the community when ya'll are ready. I think a community of maintainers is a great idea because supporting this project requires a lot of work. I might also still have ownership some of the aur repos.

**Comment by demizer** [comment]
Just want to add some encouragement, supporting this repo for six years changed my life. It did wonders for my resume and got me a few jobs. So it definitely pays off in the long run!

**Comment by woodsb02** [comment]
> Howdy folks. I started this repo many years ago. I still have ownership and can transfer that to the community when ya'll are ready. I think a community of maintainers is a great idea because supporting this project requires a lot of work. I might also still have ownership some of the aur repos.

This is great news - do you also own archzfs.com so that the package server can remain at http://archzfs.com/archzfs/x86_64/?

**Comment by endreszabo** [comment]
Given the recent developments, I also wonder:

- Where should I get the ZFS package repo db files from,
- Would it still be appropriate to host the dependent-kernel-package-referencing-repositories under kernels.archzfs.com?
- It'd also be nice to have these repo DBs regenerated automatically when new kernel versions are supported. Currently, my script polls the zfs package db repo files every hour.

**Comment by kerberizer** [comment]
@demizer I'm so glad to hear from you after all these years! It's a relief to have you around too. :)

I think we're set up well on GitHub, and the next steps would be to gradually add more maintainers (and eventually additional owners). This organization was created precisely because you were concerned that if the repos were in your personal account—as they initially were—they might need to be forked if you became unavailable to maintain them. That was a wise move and why we're able to continue the work here even while minextu is away.

If you have time to take a look at #551, it would be much appreciated. If our priority is to get the repos back online quickly, we can probably merge it now and refine it later. This would mean making GitHub the new main repository, with which the mirrors would sync using @Doridian's script (https://github.com/archzfs/archzfs/issues/545#issuecomment-2395610956).

Regarding AUR, it seems that minextu and @archzfs-bot are co-maintainers. While archzfs-bot may still function, if we don't have access to its account, we might find ourselves limited again. So perhaps it would be best if you could file an orphan request to reclaim ownership of the AUR packages. Since you were the original submitter, the process should be easier and faster. We could then add more (human) co-maintainers for redundancy.

Regarding the `archzfs.com` domain, following up on @woodsb02 and @endreszabo's questions, it's indeed interesting if you have access to it. However, personally, I still strongly lean toward moving the CI and main repo here on GitHub, as implemented by @Doridian in #551. Managing a separate domain and CI infrastructure adds complexity in terms of access control and ongoing payments, which could eventually lead to the same issues we’re dealing with now.

One important point that hasn't been discussed, I think, is the original PGP key used to sign the repo and packages. It seems to reside only on the buildbot system. If we can’t access it, we should consider generating a new key.

To avoid a single point of failure, I might decide to follow a model similar to Arch Linux’s, where a keyring contains PGP keys from multiple developers. This way, even if one keyholder becomes unavailable, the remaining developers can continue signing, and users won’t need to import a new key into their pacman keyrings.

There are a couple of other things I think are important to discuss:
- **Licensing**. Only the `archzfs-ci` repo has a LICENSE file. `archzfs` simply states in the readme that the license it MIT, but I think it would be best if we add a proper LICENSE file in all repos.
- **ZFS-kernel compatibility**, e.g. #538. We could probably maintain two repos: `stable` for packages supporting kernel versions officially compatible with the latest ZFS release, and `latest` (or `unstable`) for newer kernels—using ZFS’s master branch and/or latest release with kernel patches applied. This would give users the choice between stability and bleeding-edge updates, depending on their needs. For the stable repo, we may also need to provide older kernel packages when Arch Linux updates to newer kernels.

**Comment by kerberizer** [comment]
> * Would it still be appropriate to host the dependent-kernel-package-referencing-repositories under kernels.archzfs.com?

Good question, @endreszabo!

Personally, not only I think it's appropriate, but it's actually important to keep it. However, like with the ZFS packages repo, personally, I’d prefer to see the main 'kernel compat' repo hosted here on GitHub for resilience.

I'm concerned that the subdomain would vanish if the main domain is not renewed by minextu—assuming he has sole access to it. There was indeed a suggestion to apply for the domain and hope we'll be the first in the queue when it expires. It's probably a good idea to not lose it. I just would prefer that it isn't the single point of failure.

Depending on the registrar, of course, several people might have shared access for configuration and billing. But it means that when new people become maintainers, we'll need to remember to also share with them this access—and they should be willing to pay for the domain if needed. Likewise, when developers become inactive—particularly when they simply go silent like with minextu—we'd need to remember to check if enough active people still have access.

And we will already need to manage access for this GitHub organization, the PGP keyring, and potentially for a shared bot account on GH (unless we can replace it with some simple webhook). I'm afraid that the more credentials we'd need to keep track of to be shared properly, the more likely we'd end again in the current situation.

That's just my personal opinion, of course.

**Comment by kerberizer** [comment]
@endreszabo Would you actually be interested in joining the organization and setting up the repos here?

I would really appreciate it, as I think this is pretty important aspect of the support that we'd have to do, particularly if we want to provide a [`stable` repo](https://github.com/archzfs/archzfs/issues/545#issuecomment-2396674949) as well, with only officially supported ZFS-kernel combinations.

**Comment by endreszabo** [comment]
> @endreszabo Would you actually be interested in joining the organization and setting up the repos here?
> 
> I would really appreciate it, as I think this is pretty important aspect of the support that we'd have to do, particularly if we want to provide a [`stable` repo](https://github.com/archzfs/archzfs/issues/545#issuecomment-2396674949) as well, with only officially supported ZFS-kernel combinations.

Sure, I like the idea of keeping the building blocks together.

**Comment by kerberizer** [comment]
@endreszabo Great, thank you very much! I've invited you both here and on GitLab.

Please don't hesitate to ping me if you encounter any problems or if I could assist you in any way.

**Comment by kerberizer** [comment]
Everyone, the discussions are also enabled now if you feel like talking about topics that are not strictly issues :) https://github.com/orgs/archzfs/discussions

**Comment by travishegner** [comment]
I'm a long-time user of this project, but have built my own work around when the system is unavailable. I'm happy to see that the work on this project is moving forward, and am excited to see what the community does with it.

I did want to share my podman based build scripts here, in case they help anyone in the short-term, or in case the work is useful as a template for the new GH Actions based build system. Since it's mostly been for personal use, the work is not extremely well documented, but should be rather straightforward to follow along with. I recently switched over from regular builds to LTS builds, so it technically only supports LTS, but it wouldn't be a stretch to turn that into a flag for the build script.

Feel free to have a look here: https://github.com/travishegner/buildzfs

Also, feel free to use any of this work to carry this project forward, though I'd love credit or a shout-out for anything that you choose to use.

I'm also fairly experienced in this type of work, so if the community needs help with any of it, I'd be happy to pitch in. Just let me know what's needed.

**Comment by kerberizer** [comment]
@travishegner Thank you very much for sharing this—it does sound really helpful.

I'm thinking if we maybe shouldn't keep this org focused solely on the packages repo and the associated build scripts, but instead expand it into an umbrella for different projects related to running ZFS on Linux. This could include different build solutions that people could use on their own systems, too. In fact, I should probably upload the latest version of my own build script that I had used for many years until recently.

In any case, if you'd be interested in joining the organization and maybe even bringing this project here, you'll be more than welcome—just let me know and I'll send you an invite.

It's really great to have more people on-board, and not only for redundancy, but also for the benefits of collaboration.

**Comment by l-vanni** [comment]
@kerberizer I'm a sys/net admin and i know (and use) CI/CD systems for orchestrate my infrastructure. I could collaborate, if necessary, to project  in this context or other. I will very proud to actively collaborate on archzfs project. (my first on GitHub)

**Comment by travishegner** [comment]
@kerberizer, I'd be glad to move my build scripts into the organization and manage them there. It might motivate me to expand them a bit and perhaps allow GH actions to create the packages as another option for users to install from.

**Comment by Doridian** [comment]
> @demizer I've added you as co-owner of the GitLab organization as well.
> 
> People were asking about a backup solution and [GitLab seemed like a good option](https://github.com/archzfs/archzfs/issues/545#issuecomment-2395061565). I guess it's not a priority to set up CI there, but still nice to have the repos mirrored authoritatively somewhere outside of GitHub.
> 
> I've also invited @Doridian if they would be interested in continuing their outstanding work there as well.
> 
> These invitations—for new people who would like to start contributing to the project—are open to all of course.
> 
> Everyone, please remember that contributing isn't just about code. If you're not comfortable with writing code, you could still help with managing issues and pull requests. That's really valuable and not any less important help!

I'll take a look at GitLab. As most of the PR I've written is shell code and docker images, it should be fairly easy to convert over.
In order to make this smoother, I have changed my PR to move the Dockerfile to the root into `build-container`, so it doesn't look out of place used for GitLab as well (as there's no reason not to), and also described it more accurately.


Also reading previous posts, I agree kernel version checking for rebuilds would be great as right now the PR just runs a build once per day, with the idea being that kernel releases are not so frequent this should become an issue initially, until a better solution is designed and added ontop.

**Comment by kerberizer** [comment]
@travishegner @l-vanni Thanks for joining! If you need assistance or encounter any problems, don't hesitate to ping me.

@Doridian It's really great that you're so enthusiastic about it! And I definitely would like to see your work merged here as soon as possible, so that we can resume providing packages for everyone who relies on them.

How do you think it would be better to proceed: are there still important pieces we might want to change or add, or it should be good enough for now and we can enhance it on the go later?

I've been thinking of two possible issues:

The most important I've mentioned earlier: the PGP keys. Here's how I imagine it:
- One of us generates a new package signing key and adds it as GH secret to be used in CI.
- All developers -- you, @demizer, @endreszabo, @travishegner and myself -- sign this key with their own PGP keys.
- We announce the developers' PGP keys like https://archlinux.org/master-keys/ (could be much simpler of course)
- We could probably provide an `archzfs-keyring` package with said keys, though I have to check if populating `/usr/share/pacman/keyrings` with additional keyrings is possible beyond the default `archlinux` keyring. Of course, we'll also face the chicken-and-egg problem, as this package also needs to be signed by some key (so we'll need some kind of bootstrap procedure, requiring the users to fetch at least one PGP key directly with `pacman-key --recv-keys`). On the other hand, it could make updates for existing users more streamlined when we add or remove developers.
- Unless I'm missing something (writing really off the top of my head), this approach should be pretty resilient. As long as at least one developer remains around, changing the package signing key should be transparent, as all that is necessary for it is to be signed by one of the developers' "master" keys.

Second issue is with the possible `stable` vs `latest` or `unstable` repos as discussed [here](
https://github.com/archzfs/archzfs/issues/545#issuecomment-2396674949) (at the bottom). I think we don't necessarily need to decide now anything more than the name of the latest/unstable repo.

I'm probably fine with `latest`, though `unstable` may convey more clearly that these kernel versions are not (necessarily) officially supported by upstream ZFS, and thus pose certain risk, including for irreversible data loss.

**Comment by Doridian** [comment]
> or it should be good enough for now and we can enhance it on the go later?

Personally I think it is "good enough", for sure not perfect and not where we want to stop. But it should get a pacman repo up and going

**If other people do have concerns, I am not a master of arch packaging, so I might miss important things!**

> I'm probably fine with `latest`, though `unstable` may convey more clearly that these kernel versions are not (necessarily) officially supported by upstream ZFS, and thus pose certain risk, including for irreversible data loss.

`latest` to me conveys the idea of this being the "suggested" build, so for unstable, I'd suggest, well, either `unstable` or `experimental` or `staging` or another term that actively and unambiguously indicates that the build has the (theoretical) potential to fry your entire zpool and has not undergone whatever validation the OpenZFS devs perform to announce their compatible ranges.

So maybe just go with `stable` (or no suffix) and `experimental`?

> The most important I've mentioned earlier: the PGP keys. Here's how I imagine it:

As for your "key signing procedure", seems fine to me. And yeah, I think just taking the "top part" of the arch keys site, the `Master Signing Keys` section would be fine.

Unless someone really wants to make the pretty graph, I welcome pretty graphs usually.

**Comment by Doridian** [comment]
> Wow, that escalated quickly over the past 48 hours - in a positive way. I gave the new repo version a try in a vm - and at least to my tests with standard, lts, zen and hardened at least all loads and is able to access the test pool. So the packaging works. I also tested yay as AUR helper - and didn't got any issues. Greate work! I guess my builds are obsolete now. btw: As 2.3.0-rc1 was released - any chance to include that in the build setup - or maybe setup its own until 2.3.0 is announced official? I got it to build but had to hack it a bit as Arch doesn't like the "-rc1" part as package version.

I think that use-case should be covered by the `unstable` (or whatever it ends up being called) release, which likely runs off of latest git head.
Maybe eventually even have 3 tiers: `release`, `prerelease` (betas/RCs) and `unstable` (git `master` pretty much) or something like that.

**Comment by kerberizer** [comment]
It's really refreshing to see so many enthusiastic people!

> If other people do have concerns, I am not a master of arch packaging, so I might miss important things!

We all learn our whole lives. :) But, indeed, could I please ask **everyone** to take a look at #551 and if you have any suggestions, comments, or ideas, don't hesitate to share them.

> latest to me conveys the idea of this being the "suggested" build

This has been my main concern too :)

I would be okay with `experimental` if people like it.

Risking to piss people off by repeating myself or being Captain Obvious, let me just recap the concept for those who might've missed the previous comments, and to avoid possible confusion:
- Each ZFS release supports certain Linux kernel versions. These are typically stated in the release notes.
- ZFS releases [tend to lag behind the kernel](https://github.com/openzfs/zfs/issues/16425#issuecomment-2305895705). For some time—sometimes up to a month or more—there might not be a suitable ZFS release to match the current `linux{,-hardened,-zen}` packages (`linux-lts` is obviously an exception).
- It's possible that the ZFS `master` branch or some pull request contains compatibility fixes. It's also possible that the latest ZFS release simply compiles cleanly against the otherwise unsupported latest kernel. ZFS developers consider such cases [potentially dangerous](https://github.com/openzfs/zfs/pull/15986) and [definitely unsupported](https://github.com/openzfs/zfs/pull/15931#issuecomment-1991435220) (if you lose your data, well, "that sucks!")
- In the `stable` repo, we'll offer only supported ZFS/kernel versions. So, even though Arch have switched to kernel `6.11`, because the latest ZFS stable release is [2.2.6](https://github.com/openzfs/zfs/releases/tag/zfs-2.2.6), in `stable` we'll only offer packages built for kernel `6.10`. To keep pacman happy, we'll also have a repo with the necessary "old" kernel packages (`6.10` in this case). I'm not sure yet if we want to add the kernel packages directly to the `stable` repo or keep them in a separate one.
- For the `latest`/`unstable`/`experimental` repo (whatever name we choose), on the contrary, we'll strive to have packages for the current Arch kernels—as long as they build cleanly or compatibility patches exist. The users of that repo should be really aware that while ZFS might work perfectly fine most of the time, the risks of data corruption or loss are definitely higher—maybe much higher—than with the `stable` packages.
- We could indeed have more package repos for, e.g. ZFS beta and pre-release versions, as well as for our own testing purposes, akin to Arch's `{core,extra}-testing`. In fact, we could publish the packages for unsupported ZFS/kernel combinations first to this repo, and only after they've been tested by some people, we could push them to `latest`. Thus `latest` would still be more risky than `stable`, but less than if no preliminary testing had been performed.

As always, these are just my two cents and I welcome the discussion and exchange of ideas.

**Comment by Doridian** [comment]
For the kernels, I'd say we should have one central "kernel" repository that just checks which kernel packages we need to mirror for which sub-repo and does so.

Maybe just mirrors every released Arch kernel and drops versions less than what we need (so it would keep kernels all the way from newest until the newest build that supports ZFS, and drop all older). That way we could always be as up to date with mainline as possible, while still clearing out storage for definitely unnecessary kernels.

**Comment by kerberizer** [comment]
> In reply to https://github.com/archzfs/archzfs/issues/545#issuecomment-2397878649

The reason I thought we might put "compat" kernels in `stable` is that the users of that repo will inevitably need these kernels at some point. So, instead of telling them "if you use `stable`, always add `kernels` too", we could just provide the kernels from the same repo. And if people forget to also add `kernels` to their `pacman.conf`, there will be rants about "your packages broke my installation, you ..." :D

There could be downsides, of course, or outright deal breakers. I personally would be fine with a separate repo as well, especially if there's a consensus that this would be the best approach. One reason why a separate repo or even repo**s** might be needed, I guess, is https://end.re/blog/ebp036_archzfs-repo-for-kernels/#update-on-2020-12-17

**Comment by Doridian** [comment]
> The reason I thought we might put "compat" kernels in `stable` is that the users of that repo will inevitably need these kernels at some point.

Yeah, it would break updating until people add those repos. Well, with pacman's `-Syu` behaviour anyway.
But if you exclude linux/kernel packages from the upgrade, you can still keep your install mostly up to date (and also exclude things like nVidia drivers, that depend on a fixed kernel version like ArchZFS). That's the way I currently run my ZFS-bearing systems

But I think since we have to tell people to change repos anyway (if we want the GitHub one to be primary, or figure out some mirrorlist system unless such already exists), making sure the `kernel` repo is well documented might suffice?

**Comment by Soulsuke** [comment]
Hello @kerberizer , if I can add my two cents: I've sort of pushed zfs in my workplace (root snapshots are life savers), and I can attest adding a separate repository causes a good deal of headaches for many users... I've had to set up a (really stupid) script to check the newly released archzfs packages, fetch the relative kernel versions and then add them in a custom repo to minimize issues.

I'd like to point out a few key points of such process hoping to be helpful:

- you'll also need to provide the headers, or dkms will break
- I've dropped support for the -git packages, as they tend to depend on the same kernel but with a different version (which isn't supported on the same repo afaik)
- it should be written in the doc that this counts as a partial update, as we may lag behind the official repos.... Which becomes annoying if a new gcc version is released and dkms modules fail to build because of a compiler version mismatch (which can be solved downgrading + ignoring gcc updates)

**Comment by SoongNoonien** [comment]
> ZFS releases tend to lag behind the kernel. For some time—sometimes up to a month or more—there might not be a suitable ZFS release to match the current linux{,-hardened,-zen} packages (linux-lts is obviously an exception).

While `linux-lts` definitely breaks (way) less often it has become incompatible with ZFS in the past. This is mainly due to Arch being "bleeding edge" even when providing lts packages. Sometimes the kernel maintainers announce very early that a new kernel will receive lts support. Usually as soon as this is the case Arch will update their `linux-lts` package. This lead to the bizarre situation in the past that the `linux` and `linux-lts` packages had the same kernel version.

> In the stable repo, we'll offer only supported ZFS/kernel versions. So, even though Arch have switched to kernel 6.11, because the latest ZFS stable release is [2.2.6](https://github.com/openzfs/zfs/releases/tag/zfs-2.2.6), in stable we'll only offer packages built for kernel 6.10. To keep pacman happy, we'll also have a repo with the necessary "old" kernel packages (6.10 in this case). I'm not sure yet if we want to add the kernel packages directly to the stable repo or keep them in a separate one.

What does "old" kernel packages mean exactly? Right now 6.11.2 is the newest kernel in the Arch repos (and in general) and 6.10.13 is the newest kernel supported by ZFS but 6.10.13 never released for Arch. The newest version of the 6.10 kernel released on Arch was 6.10.10 three weeks ago. I'm not sure if it is a good idea to "suggest" using unsupported kernel versions. Of course ArchZFS could build their own 6.10.13 with the Arch config but maintaining this would only increase the burden.

**Comment by kerberizer** [comment]
@SoongNoonien Both are good points, thank you.

Well, unless we want to tell people to just "deal with it", maybe we could do this:
- Track Arch's kernel package repos here.
- When Arch switches to a new kernel, temporarily stop tracking their repos and instead start updating the kernel patch version directly from kernel.org.
- Simultaneously, also start building these packages ourselves, instead of simply fetching them from Arch.
- Once a compatible ZFS version for the newest kernel is released, return to the usual tracking and fetching.

These switches happen when the "previous" kernel is already more or less mature, so, for example, by the time Arch adopts 6.11, it's unlikely that newer 6.10 versions would need any (new) Arch-specific fixes.

This means we could probably fully automate the whole process.

Of course, in extreme cases it could happen that the "previous" kernel reaches EoL before ZFS catches up. But I think we can live with this. And for LTS—which is the safer approach for ZFS anyway—the likelihood of this happening is zero.

**Comment by kerberizer** [comment]
> * it should be written in the doc that this counts as a partial update, as we may lag behind the official repos.... Which becomes annoying if a new gcc version is released and dkms modules fail to build because of a compiler version mismatch (which can be solved downgrading + ignoring gcc updates)

@Soulsuke that's a very good point (the rest is really helpful too) and important to keep in mind.

The `stable` repo is obviously going to be tricky and require a lot more work to create and maintain.

**Comment by SoongNoonien** [comment]
> ...it's unlikely that newer 6.10 versions would need any (new) Arch-specific fixes.

I would think so too.

> This means we could probably fully automate the whole process.

That would be nice.

> Of course, in extreme cases it could happen that the "previous" kernel reaches EoL before ZFS catches up.

I'm not sure if this is actually this unlikely. From what I remember this happens from time to time.

> But I think we can live with this.

Yes, if someone has a problem with this they could switch to LTS.

> And for LTS—which is the safer approach for ZFS anyway—the likelihood of this happening is zero.

I'm using LTS for about five years now. During this period there was always a ZFS compatible LTS version available at kernel.org. But if I'm not mistaken when the kernel maintainers started to mark symbols GPL-only which are used by ZFS they considered this a minor and non-breaking change. This had the effect that for some time no up to date kernel was compatible with ZFS including all LTS kernels. This was more than five years ago and I think that this shouldn't happen again.

**Comment by Soulsuke** [comment]
Would building the kernel be that useful though? It'd still risk to break as many things as keeping the last supported kernel would... Of course it would be great to have if certain scenarios where severe mitigations get backported, but I fail to see any ither advantage.

I also wonder if providing the whole kernel package signed with a non-official Arch key may end up scaring away new users or rise concerns about security...

I mean, it'd be great to be independent from Arch's official kernel releases, but is the actual effort be worth it?

**Comment by lnxsrt** [comment]
> Would building the kernel be that useful though? It'd still risk to break as many things as keeping the last supported kernel would... Of course it would be great to have if certain scenarios where severe mitigations get backported, but I fail to see any ither advantage.
> 
> I also wonder if providing the whole kernel package signed with a non-official Arch key may end up scaring away new users or rise concerns about security...
> 
> I mean, it'd be great to be independent from Arch's official kernel releases, but is the actual effort be worth it?

Is there a way to create a virtual package in Arch? Could create a virtual kernel package that would point to the correct official kernel version that ZFS was compiled against.

If not, either option isn't great. Right now I have a script that checks the kernel version for the zfs package and updates the two in tandem. Works most of the time, but cumbersome.

**Comment by Soulsuke** [comment]
@lnxsrt with one repo hosting both zfs and the latest supported kernel, it'd take precedence over the official repos if you put it before them in pacman.conf... That'd solve the issue, right?

@SoongNoonien I agree with your reasoning from a dev point of view, lts causes less hassles... But from a user's pov, I'd rather switch to dkms and risk partial updates, newer kernels tend to come with very nice improvements om the gaming side...

IMHO supporting lts shouldn't be in question as it's the solution that causes less headaces, but we should also support the main kernel aling with a big "USE AT YOUR OWN RISK" disclaimer printed out when installing the zfs package, maybe linking to a pinned discussion?

**Comment by demizer** [comment]
> > Howdy folks. I started this repo many years ago. I still have ownership and can transfer that to the community when ya'll are ready. I think a community of maintainers is a great idea because supporting this project requires a lot of work. I might also still have ownership some of the aur repos.
> 
> This is great news - do you also own / have edit access to the DNS records for archzfs.com so that the package server can remain at http://archzfs.com/archzfs/x86_64/?

I transferred the domain to @minextu. Sorry!

> @demizer I've added you as co-owner of the GitLab organization as well.

Thanks @kerberizer but this is not necessary. While I still use arch, I do not use ZFS on Arch. I unfortunately do not have enough time to contribute to Arch due to family and work obligations.

**Comment by demizer** [comment]
I also can't help much with Github actions since I don't have much experience there. I can help review Gitlab pipelines though since that is part of my current employment.

**Comment by lnxsrt** [comment]
> @lnxsrt with one repo hosting both zfs and the latest supported kernel, it'd take precedence over the official repos if you put it before them in pacman.conf... That'd solve the issue, right?

Ah, I see what you are saying. Just duplicate the official kernel package on the zfs repo. It would still be the official Arch signed kernel pacakge. That would work too.

**Comment by johnramsden** [comment]
I might be late to the party in offering this but I would love to assist in any effort if there is work that is needed. But I'll admit I i'm not sure where I can be of help. Could someone let me know if there's any way I could help assist in the effort? I have knowledge of scripting, infrastructure (if buildbot help is needed) as well as automation if assistance is needed with actions.

**Comment by kerberizer** [comment]
@johnramsden Not late at all! :)

It's really so nice to see people who are enthusiastic to help! I've sent you an invite. You could maybe take one final look at #551 before we deploy it, as I haven't seen comments from other people and another pair of eyes is always helpful.

**Comment by kerberizer** [comment]
@ZLima12 `linux-lts-hardened` is a very good idea. Coincidentally, I've started assembling my own build a few weeks ago, after realizing that what I've (mis)read a while ago about `linux-lts` having the same hardening features as `linux-hardened` wasn't true. What do you think about setting up a repo here for the sake of collaboration? I think it still fits into the scope, especially with a future `stable` repo, considering that there are some of us crazy enough to run production servers on Arch Linux. Joking aside, if you're interested, please ping me and I'll send you an invite to the org.

**Comment by kerberizer** [comment]
Speaking of the repos, I'm so grateful to **everyone** who has been sharing opinions. Some very interesting and important points have been raised, and while this discussion is probably far from over, I think we're having a great start.

I'll see if I can summarize it in a new thread, so that we can continue it (I do apologize for being mostly silent the last few days, but it's been a tough week and I managed to catch some cold on top of it).

One thing stood out to me in particular, by the way: the very real risk of developer burnout.

Too wide scope, too ambitious goals, and not enough resources can easily kill any project—but particularly so volunteer projects. So starting with modest goals and expanding from there in small steps is probably a good idea.

Let's really start with our first repo from #551.

From all that has been discussed so far, I think that I like @Doridian's idea to call it `experimental` more and more.

Not only currently it builds rather "bleeding edge" packages, but we will, after all, experiment with it how this completely GitHub based approach will work.

**If you have other ideas about the name, please don't hesitate to share them, but maybe let's target Monday for merging, so that people can start testing it next week.**

I still haven't completely settled on the PGP signing system and I haven't seen many comments about it, so maybe in the beginning we could test with `SigLevel = Optional`.

The repo and packages will actually be signed with the key I've generated for the project, but the question remains about how the PGP trust will be organized.

My idea remains that several developers sign this key with keys of our own—and it is these keys that we actually announce as "trusted keys". This would allow the actual signing key to be replaced whenever necessary.

I've already checked that [pacman keyrings can indeed be distributed as packages](https://man.archlinux.org/man/pacman-key.8#PROVIDING_A_KEYRING_FOR_IMPORT) beyond the default `archlinux` keyring, so we could likely use this to have the developer keys kept updated for existing users.

For bootstrapping with new users, we could have a dedicated `pgp-keys` repo here with tight security permissions, which would contain a convenient for importing dump of the public keys of the trusted developers, plus probably some basic information about the keys like the top of the page here: https://archlinux.org/master-keys/

If you have any ideas about this, please don't hesitate to share them!

**Comment by jbrodriguez** [comment]
@kerberizer what will stable, experimental mean ?

- `stable`: latest linux-lts + latest openzfs (highly unlikely that openzfs won't support latest linux-lts)
- `experimental`: latest linux + latest openzfs (thay may or may not support latest linux?)

**Comment by kerberizer** [comment]
@jbrodriguez Yes, that's a good approximation, in my opinion.

If you'd ask me as purely a user, who might be unaware of the resources and complexities involved in maintaining these repos, I'd probably answer that I want to see:
- A `stable` repo for critical applications, people requiring very high stability, etc., which would provide only well tested combinations of ZFS/Linux versions, most likely, indeed, targetting only `linux-lts` (and possibly variants like `lts-hardened`).
- A `bleeding-edge` or `experimental` repo for those who want the very latest features and are willing to take the associated risk of data corruption or loss.
- Perhaps a 'middle ground' repo that provides ZFS support for the latest Arch Linux kernels, but only after compatible ZFS releases are available. This wouldn't be without risks (e.g., from being forced to use an older kernel at times), but at the same time could be both less risky than `bleeding-edge`, and more featureful than `stable`.

**Comment by Soulsuke** [comment]
Just to clarify, what would be the difference between experimental and middle-ground? Would the former target zfs-git against the current arch kernel, while the latter would target actual zfs releases and the latest arch kernel it supports? 

For the record, I can help testing out a repo using official zfs releses targeting arch kernels.

I'd also volunteer for the project, but I'm not sure I could really help out as I have no real experience with ci/cd

**Comment by l-vanni** [comment]
In my opinion is very important to build last openzfs version with last released kernel (only if it's supported by zfs version). To check compatible version, we could add in `conf.sh` the last kernel supported in specific openzfs release before build it.  This is called `stable`. OK for `experimental`,`testing` or `unstable` for not 100% compatible version. (use at own risk!)

**Comment by jbrodriguez** [comment]
> Edit: `linux-lts` branch called `longterm` not `stable`
ooo right, you have a point there, this would somewhat align with kernel naming conventions

**Comment by kerberizer** [comment]
> Just to clarify, what would be the difference between experimental and middle-ground? Would the former target zfs-git against the current arch kernel, while the latter would target actual zfs releases and the latest arch kernel it supports?

Yes, mostly this. Besides `zfs-git`, `experimental` would also provide provide patched ZFS releases and beta/RC versions like https://aur.archlinux.org/packages/zfs-dkms-staging-git (BTW, does anybody know the developer behind it?)

> I'd also volunteer for the project, but I'm not sure I could really help out as I have no real experience with ci/cd

Even just monitoring issues and discussions and triaging them can be an invaluable help. I've sent you an invite.

**Comment by jbrodriguez** [comment]
there are only 2 hard problems in computer science: naming things, cache invalidation and off-by-one errors 🤣

**Comment by justinfenn** [comment]
Hi @kerberizer, managing the signing keys as a keyring package is a fine approach. I do this with a private repo, and it works really well. I basically just follow the structure of the [Arch keyring repo](https://gitlab.archlinux.org/archlinux/archlinux-keyring) and use that tooling.

Bootstrapping is an issue. The easiest way I've found to deal with it is to initially install the package using `pacman -U` with some manual verification. After that it's really easy.

**Comment by l-vanni** [comment]
> Some people raised concerns that `stable` can be somewhat misleading for such repo, because people will be stuck with older and unsupported kernels while we wait for ZFS to come up with a release compatible with latest Arch kernels.

Ok, that's true, very true. But people should remember that previously this was the normal (aka: `stable`) way of building package. To recap: (in my opinion - my mind)

- `stable` (or another name, for example `supported` or `tested`) repo match last kernel supported with last ZFS version (non RC);
-  `unstable` last ZFS version (non RC) with last arch official kernel;
-  `experimental` ZFS RC version with last arch official kernel;
- `git` replicate ZFS master branch with last arch official kernel;
- `longterm` ZFS lts with `linux-lts` kernel (safe, supported, production-ready)

This approach is focused to ZFS and not for kernel. Only risk (serious) to run package in `stable` repo isn't to run last version of arch supported kernel. It's a risk obviously but system not have compatibility problem.

The users should decide which repo use after read README.

> But otherwise this is exactly the repo that I've been calling "middle ground" above and I agree that it's a good idea to have it at some point. I don't know if some other name might be more appropriate, though, like `mainline`.

`mainline` sounds good!

> Maybe `longterm` can be yet another repo, that would target **both** `linux-lts` and the [latest ZFS LTS release](https://endoflife.date/openzfs)?

It's a very good idea!

**Comment by n0xena** [comment]
My 2 cents (just personal opinion - to be ignored):
`latest`: latest ZFS non-RC release with latest supported kernel - as of today: 2.2.6 with 6.10.10 from september 12th
`unstable`: latest ZFS non-RC release with latest Arch kernel - as of today: 2.2.6 with 6.11.3
`experimental`: latest ZFS release including RC with latest Arch kernel - as of today: 2.3.0-RC1 with 6.11.3
`stable`: latest ZFS non-RC release with latest Arch LTS-kernel - as of today: 2.2.6 with 6.6.56
`git`: well - current git master
additional as needed like zen, hardened, dkms, ...

As it was mentioned many times: As Arch and its fast pace rolling release the only "stable" option to use zfs in a supported manner will be latest non-RC on LTS - hence a branch called `stable` should reflect both: The latest official ZFS release combined with the latest Arch kernel supported by the given ZFS release. If I recall correct for 2024 it not happened that the versions of the latest zfs release and the current arch main standard kernel matched - and if so only for a few days rather by coincidence.
Just because the current 2.2.6 theoretical works with 6.11.x (it does practical - I run it right now) it's a not official supported configuration and by this should not called `stable`. People should imply `stable` is the current active official supported combination of both ZFS and Arch - if this ends up to be LTS so be it.
There's also the 2.1.x line in active development. Although I personal don't understand the differences and if stuff gets backported as it's still an active branch yet on an older codebase it might should be actively supported as well?

The idea of the keyring sounds really nice. This way not a single person is responsible but the project as an entity itself with its maintainers backing the master key. While I personal would love some PKI style PGP is based on the idea of web-of-trust and hence inherits this "there's no authority" principle.
Another idea to this would be: As archzfs is a project specific for Arch why not have its keyring signed by the arc-keyring so there comes up an explicit chain of trust: One who trusts Arch would implicitly trust archzfs as its master key is trusted by the arch-keyring. I see it equivalent to if archzfs would be elevated into the official extra repo. And, who knows, maybe in the future when this licence nonese was figured out ZFS maybe could become part of upstream kernel mainline?

Anyway - these are just my "sunday afternoon thoughts" to be ignored.
I really love the momentum this project gained over the past few weeks.
ZFS is such an amazing tool: It combines RAID-like low-level device management with it's own I/O path along with a filesystem with built-in features like copy-on-write, block-level reference and de-dup, snapshots, sub-datasets and -volumes and so much more. Combined with Arch as one of the main base distributions it's a good option to base your storage on.
And looking over at https://github.com/openzfsonwindows/openzfs it's a great candidate for a new standard cross-OS filesystem.

@minextu Whatever the reason might be - I hope you do well and come back soon. You did a greate work. We miss you.

**Comment by nerzhul** [comment]
regarding @n0xena topic, i'm in favor of merging latest with stable. Tbh when i see that more recent than 6.10.10 are published i expect pacman to fail as it's not compatible with ZFS and i put linux package in my pacman.conf ignorepkgs, like currently. If a new package supports a upper version i remove the line and let's go for kernel upgrade.

**Comment by johnramsden** [comment]
What is the plan moving forward since we now have builds working thanks to the good work in #551. I see experimental was released, are we planning on leaving it for a bit to make sure there are no issues and then moving forward with the other builds?

**Comment by johnramsden** [comment]
Since we now have the experimental release should I go ahead and update the wiki to represent the new repository? Or is it too early for that

**Comment by woodsb02** [comment]
Suggest it would be best done in 2 steps - add the details of the new experimental repo now, and then later remove the old archzfs repo from the wiki and remove the "experimental" tag from the new repo.

**Comment by SebastianS90** [comment]
Is it guaranteed that the current `experimental` release on GitHub only contains `zfs-linux*` packages for kernel versions that are officially supported?

If not, then I'd suggest to first implement a release that has such a guarantee before updating the wiki, or at least put a strong warning there to not blindly perform kernel updates when using `experimental`. Otherwise users might update their system to an unsupported kernel version and corrupt their pool.

**Comment by Soulsuke** [comment]
I'm currently testing the repo on 4 units and it's behaving well so far, but I agree we should wait for a repo that targets the latest officially supported kernel, as afaik 2.2.6 still has a few issues on 6.11 which will be fixed in 2.3 (although it's nothing critical)

**Comment by n0xena** [comment]
I have to stress that the last 2.2.6 is not fully stable on 6.11.x for now: coincidently I tested it while my pool failed and had to rebuild several drives
I can't tell if the failures were caused by 2.2.6 on 6.11.x but the rebuild sure wasn't fun
please consider to either stick to 6.10 with 2.2.6 or switch to lts 6.6.x until 2.3.0 is released with proper 6.11 support
btw I also have test run with my own build of 2.3-rc1 on 6.11 - and let me put it this way: no matter it was in a vm with virtual drives - it was even less fun - highly unstable - try it at your own risk!

**Comment by ThatWileyGuy** [comment]
I can report that I've been running 2.2.6 on 6.6.56 and it's been delightfully boring. :)

**Comment by stevleibelt** [comment]
Thanks @n0xena for your report and your work upfront to be able to report. LTS 6.6 should do it until december 2026 so :crossed_fingers: that openzfs will fix the issues.

**Comment by khenderick** [comment]
I just noticed a manual update on https://aur.archlinux.org/pkgbase/zfs-linux that has pushed a 2.2.6 on 6.11.4 which was described as unstable on this thread a few posts earlier. Did recent ZFS changes on 2.2.6 made it compatible (and can I safely update), or is the zfs-linux package now experimental (which I think should be clearly stated since that wasn't the case before if I understand correctly).

**Comment by n0xena** [comment]
@khenderick please see https://github.com/openzfs/zfs/discussions/16574
currently 2.2.6 "works" on 6.11 as it is "technical" compatible - but official support is planed for 2.2.7/2.3.0

**Comment by lightdot** [comment]
I have taken over the zfs-linux AUR package and downgraded it to use the 6.10 kernel, as this seems the proper thing to do at the moment.

I'd gladly contribute to this project too, as much as I can. I have been following the discussion here and everything seems to be going in the right direction, nothing much to be added to what has already been suggested and done...

**Comment by ArnCo** [comment]
> The AUR version is also missing the hooks to allow unlocking an encrypted root pool via ssh.

I contributed by implemented this PR and I approve this message. I have rolled back to aur/zfs-dkms for my laptop, but still running the archzfs dkms package on my desktop so that I can unlock it remotely.

@lightdot is there any way that we could integrate the zfsencryptssh hook to the aur package?

**Comment by endreszabo** [comment]
> missing the hooks to allow unlocking an encrypted root pool via ssh

As a side note, let me introduce you guys to the excellent [zfsbootmenu](https://docs.zfsbootmenu.org/en/v2.3.x/) project, which also supports remote unlocking via SSH and has many, many more ZFS features while still being a bootloader. I use this to remote-unlock my bare-metal headless server running at Hetzner.

**Comment by lightdot** [comment]
@ArnCo I see that this hook is a part of the zfs-utils package in AUR. I'm not familiar with its usage, what changes are needed for the zfs-linux package in AUR?

**Comment by n0xena** [comment]
@endreszabo would be cool if the one responsible for publishing the releases could include the versions of zfs and supported linux kernels - and think about versioning, as the 2.3.0 release suggest that it uses zfs 2.3.0, which isn't released yet and the text doesn't mention any zfs or kernel version

**Comment by stevleibelt** [comment]
Other topic,

could @johnramsden maybe rename this issue to something like "future development of archzfs"?
This is more a marketing move unleashed from a person who is bad in marketing :-D.

**Comment by kerberizer** [comment]
> could @johnramsden maybe rename this issue to something like "future development of archzfs"? This is more a marketing move unleashed from a person who is bad in marketing :-D.

I've been thinking about converting this to a discussion. Maybe a general discussion topic about ArchZFS development.

Thoughts? Objections?

Meanwhile, I must apologize for being so slow with https://github.com/archzfs/archzfs-keyring/pull/1, but it isn't terribly critical right now and I'd prefer to err on the side of caution with it (and time has been tight recently).

**Comment by Soulsuke** [comment]
@kerberizer I'd say make it a discussion and maybe write somewhere we're in the middle of a little overhaul? Like on the main page and/or the wiki. I'd also close issues like #543 adding a ling to the discussion

Also... Thanks for your work on the keyring, and for kickstarting back this project to life :p

**Comment by johnramsden** [comment]
> > could @johnramsden maybe rename this issue to something like "future development of archzfs"? This is more a marketing move unleashed from a person who is bad in marketing :-D.
> 
> I've been thinking about converting this to a discussion. Maybe a general discussion topic about ArchZFS development.
> 
> Thoughts? Objections?
> 
> Meanwhile, I must apologize for being so slow with [archzfs/archzfs-keyring#1](https://github.com/archzfs/archzfs-keyring/pull/1), but it isn't terribly critical right now and I'd prefer to err on the side of caution with it (and time has been tight recently).

There was some discussion that we should hold off before "advertising" that the repository is ready which is why I did not update the wiki. If the consensus is we should notify of the experimental repo I can update

**Comment by rearden888** [comment]
> > could @johnramsden maybe rename this issue to something like "future development of archzfs"? This is more a marketing move unleashed from a person who is bad in marketing :-D.
> 
> I've been thinking about converting this to a discussion. Maybe a general discussion topic about ArchZFS development.
> 
> Thoughts? Objections?
> 
> Meanwhile, I must apologize for being so slow with [archzfs/archzfs-keyring#1](https://github.com/archzfs/archzfs-keyring/pull/1), but it isn't terribly critical right now and I'd prefer to err on the side of caution with it (and time has been tight recently).

I definitely think we should convert/move this to a discussion.  That's certainly more appropriate for what is being discussed.
